### PR TITLE
Add .zfill() Method Entry in 'Terms' Section (#3367)

### PR DESCRIPTION
**Pull request by KimFarida**
### Description
Added a new entry for the '.zfill()' method in the 'Terms' section. Created a new markdown file named 'zfill.md' under the zfill folder with usage examples and explanations for the method.
This change closes issue #3367.

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR

### CODE REVIEW

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+---
+Title: '.zfill()' 
+Description: 'Takes an int and returns a copy of the string with a number of zeros preppended as a suffix , i.e "no. of zeros" + string.'
```
Comment (SSwiniarski) [inline content/python/concepts/strings/terms/zfill/zfill.md:3]:
```suggestion
Description: 'Takes an int and returns a copy of the string with a number of zeros prepended as a prefix, i.e "no. of zeros" + string.'
```

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/analyze-data-with-python'
+---
+
+The **`.zfill()`**  method pads a string with zeros on the left to maintain a specific length. It takes an integer argument, and the number of zeros added is determined by the difference between the specified length and the length of the original string.
+
+> **Note:** The `.zfill()` method does not change the string it is used on.
+If no argument is passed into the method. A type error is thrown
```
Comment (SSwiniarski) [inline content/python/concepts/strings/terms/zfill/zfill.md:19]:
```suggestion
> **Note:** The `.zfill()` method does not change the string it is used on.
```
We don't need the line about the error, just note above that the parameter is required.

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Syntax
+```pseudo
+string.zfill(arg(int))
+```
```
Comment (SSwiniarski) [inline content/python/concepts/strings/terms/zfill/zfill.md:24]:
````suggestion
## Syntax

```pseudo
result = string.zfill(arg)
```
````
Also add a sentence after the code block explaining what `result` and `arg` are.

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+## Codebyte Example
+The following example shows how `.zfill()` returns a copy of that string with leading zeros.
```
Comment (SSwiniarski) [inline content/python/concepts/strings/terms/zfill/zfill.md:27]:
```suggestion
## Example 1

The following example shows how `.zfill()` returns a copy of that string with leading zeros: 
```

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+```shell
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+TypeError: zfill() takes exactly 1 argument (0 given)
+```
+
+The above error simply means the `.zfill` method is missing the integer paramater
+
+## Codebyte Example
+The following example shows how `.zfill()` method helps ensure a specific length for a string, adding zeros to the beginning if the string is shorter than the desired length.
```
Comment (SSwiniarski) [inline content/python/concepts/strings/terms/zfill/zfill.md:80]:
```suggestion
## Codebyte Example

The following example shows how `.zfill()` method helps ensure a specific length for a string, adding zeros to the beginning if the string is shorter than the desired length.
```

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+Codecademy
+@Codecademy
+
+00Codecademy
+0@Codecademy
+000000000cat
+
+```
+
+## Example
```
Comment (SSwiniarski) [inline content/python/concepts/strings/terms/zfill/zfill.md:60]: This section is really not necessary. All functions with required parameters will throw an error if the parameter is missing. It isn't something specific to `.zfill()`

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@KimFarida I did a first review and left a few comments.

**Review by Spikke88 — APPROVED** [review]

**Comment by caupolicandiaz** [comment]
@KimFarida There hasn't been any activity on this entry, are you still available to finish this submission?

**Comment by KimFarida** [comment]
> @KimFarida There hasn't been any activity on this entry, are you still available to finish this submission?

Yes I am, sorry it took so long. 
had a few personal I had to handle so wasn't available for a while. I'm okay now
I'll put in the submission before the week ends

**Comment by KimFarida** [comment]
> @KimFarida There hasn't been any activity on this entry, are you still available to finish this submission?

I have made the required changes

**Review by SSwiniarski — APPROVED** [review]
@KimFarida looks good for a second review. (Sorry for the delay)

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+```py
+first_string = "Codecademy"
+second_string = "@Codecademy"
+third_string = "cat"
+
+print(first_string)
+print(second_string)
+
+# parameter is less than/ equal to string length, output will be the same string.
```
Comment (caupolicandiaz) [inline content/python/concepts/strings/terms/zfill/zfill.md:40]:
```suggestion
# If the argument is less than/equal to string length, output will be the same string.
```

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+> **Note:** The `.zfill()` method does not change the string it is used on.
+
+## Syntax
+
+```pseudo
+result = string.zfill(arg(int))
+```
+
+It creates a new string with leading zeros based on the specified integer argument, represented by `arg`. The modified string is stored in the variable `result`.
```
Comment (caupolicandiaz) [inline content/python/concepts/strings/terms/zfill/zfill.md:26]:
```suggestion
It creates a new string with leading zeros based on the specified integer argument, represented by `total_length`. The modified string is stored in the variable `result`.
```

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+print(second_string)
+
+# parameter is less than/ equal to string length, output will be the same string.
+print(first_string.zfill(10))
+print(second_string.zfill(10))
+
+print(first_string.zfill(12))
+print(second_string.zfill(12))
+print(third_string.zfill(12))
+```
```
Comment (caupolicandiaz) [inline content/python/concepts/strings/terms/zfill/zfill.md:47]: There should be an empty line after the code block

File: content/python/concepts/strings/terms/zfill/zfill.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+Codecademy
+@Codecademy
+
+Codecademy
+@Codecademy
+
+00Codecademy
+0@Codecademy
+000000000cat
+
```
Comment (caupolicandiaz) [inline content/python/concepts/strings/terms/zfill/zfill.md:60]:
```suggestion
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@KimFarida I've added my comments. Please update the entry, and tag me when the changes are complete. Thanks

**Review by caupolicandiaz — APPROVED** [review]
@KimFarida @SSwiniarski I've gone ahead and incorporated the edits I suggested.

**Pull request merged**